.github/workflows/subgraph.yml

```diff
@@ -0,0 +1,35 @@
+name: Subgraph
+
+on:
+  pull_request:
+    branches:    
+      - main
+
+jobs:
+
+  test-subgraph:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: subgraph
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      
+      - name: Install dependencies
+        run: yarn --ignore-scripts
+
+      - name: Run codegen
+        run: yarn codegen
+          
+      - name: Test Subgraph
+        run: yarn test
```

Comment:
What about the `build` command?

Reply:
Only needed for deployments to Hosted Service.
This PR contains commands to run Subgraph unit tests.

Reply:
But we just saw in the other PR that a change broke the build command.  Why would we want to submit changes we know break the build command?

Reply:
I think it's just that especially now that we are going to run the gh action for develop prs too, some contract change that changes the event ABIs can break the subgraph build command.

But as long as the default value for the path to the ABI is deployments, that shouldn't make a difference so I'll add the build command here too. (It will fail for the next PR because the ABI of the deployed contract will be different from the changes in the subgraph config files since we have not yet deployed the new version.)